src/iqm/qiskit_iqm/iqm_job.py

```diff
@@ -75,7 +75,11 @@ def _format_iqm_results(self, iqm_result: RunResult) -> list[tuple[str, list[str
             raise ValueError(
                 f'Cannot format IQM result without measurements. Job status is "{iqm_result.status.value.upper()}"'
             )
-        requested_shots = self.metadata.get('shots', iqm_result.metadata.shots)
+        if iqm_result.metadata.request is None:
+            raise ValueError(
+                f'Cannot format IQM result without original request. Job status is "{iqm_result.status.value.upper()}"'
+            )
+        requested_shots = self.metadata.get('shots', iqm_result.metadata.request.shots)
```

Comment:
Looks like something went wrong with some merge, the change here reverts a change done intentionally in https://github.com/iqm-finland/qiskit-on-iqm/pull/132. Should be re-reverted:

```suggestion
        requested_shots = self.metadata.get('shots', iqm_result.metadata.shots)
```